MINOR: re-enable WorkerTest for java 16+

LiamClarkeNZ opened the pull request:
…As it no longer uses Powermock, it can be run on Java 16+

Signed-off-by: Liam Clarke-Hutchinson <liam@steelsky.co.nz>

Verified by running unit test with following JVM: 

```
liam.clarke-hutchinson@ELH-5AMD6R ~/d/kafka (WorkerTestJava16)> java -version
openjdk version "18.0.2" 2022-07-19
OpenJDK Runtime Environment Temurin-18.0.2+9 (build 18.0.2+9)
OpenJDK 64-Bit Server VM Temurin-18.0.2+9 (build 18.0.2+9, mixed mode, sharing)
```

### Committer Checklist (excluded from commit message)
- [X] Verify design and implementation 
- [X] Verify test coverage and CI build status
- [X] Verify documentation (including upgrade notes)

LiamClarkeNZ committed:
Fix small regression, where WorkerTest was re-disabled for Java 16+. As it no longer uses Powermock, it can be run on Java 16+

Signed-off-by: Liam Clarke-Hutchinson <liam@steelsky.co.nz>

showuon commented:
Sorry @LiamClarkeNZ , could you explain more why we should disable WorkerTest? It doesn't fail in the latest trunk build, where jdk 8 and jdk 11 runs well.
https://ci-builds.apache.org/job/Kafka/job/kafka/job/trunk/1172/

LiamClarkeNZ commented:
Hey @showuon! No, I'm _re-enabling_ `WorkerTest` for test runs under Java 16+, not disabling it :) 

It was excluded from test runs under Java 16+ previously, because it used PowerMock, which does not work for Java 16+. But I refactored it to use Mockito some time ago, and removed it from the Java 16 exclusion list.

A later PR accidentally added it back to the Java 16 exclusion list. 

Does that make sense? :)

showuon reviewed (approved):
Nice! And yes, I re-read the code change, it is re-enabling it now. And build pass. Thanks.

showuon commented:
Failed tests are un-relabed.
Build / JDK 11 and Scala 2.13 / kafka.server.RaftClusterSnapshotTest.testSnapshotsGenerated()
